### PR TITLE
add local registry tutorial

### PR DESCRIPTION
**Pull request by BenTheElder**
xref: #110

### CODE REVIEW

File: site/content/docs/user/local-registry.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+menu:
+  main:
+    parent: "user"
+    identifier: "user-local-registry"
+    weight: 3
+---
+# Local Registry
+
+With kind v0.6.0 there is a new config feature `containerdConfigPatches` that can
+be leveraged to configure insecure registires.
```
Comment (aojea) [inline site/content/docs/user/local-registry.md]: Typo registries

File: site/content/docs/user/local-registry.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+The following recipe leverages this to enable a localregistry.
```
Comment (aojea) [inline site/content/docs/user/local-registry.md]: Local registry??

File: site/content/docs/user/local-registry.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+The following shell script will create a local docker registry and a kind cluster
+with it enabled.
+
+```
+#!/bin/sh
+set -o errexit
+
+# create registry container unless it already exists 
+REGISTRY_CONTAINER_NAME='kind-registry}'
```
Comment (aojea) [inline site/content/docs/user/local-registry.md]: Remove trailing }

File: site/content/docs/user/local-registry.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  docker exec "${node}" sh -c "echo $(docker inspect --format '{{.NetworkSettings.IPAddress }}' "${REGISTRY_CONTAINER_NAME}") registry >> /etc/hosts"
+done
+```
+
+## Using The Registry
+
+The registry can be used like this.
+
+1. First we'll pull an images `docker pull docker pull gcr.io/google-samples/hello-app:1.0`
+2. Then we'll tag the image to use the local registry `docker tag gcr.io/google-samples/hello-app:1.0 localhost:5000`
```
Comment (aojea) [inline site/content/docs/user/local-registry.md]: Previously you made 5000 a variable in REGISTRY_PORT

Comment (BenTheElder) [reply site/content/docs/user/local-registry.md]: it's not overridable though, that was just to make the script refactorable

File: site/content/docs/user/local-registry.md (hunk 3 above)
Comment (BenTheElder) [reply site/content/docs/user/local-registry.md]: fixed

File: site/content/docs/user/local-registry.md (hunk 2 above)
Comment (BenTheElder) [reply site/content/docs/user/local-registry.md]: done

File: site/content/docs/user/local-registry.md (hunk 1 above)
Comment (BenTheElder) [reply site/content/docs/user/local-registry.md]: done

File: site/content/docs/user/local-registry.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+for node in $(kind get nodes); do
+  docker exec "${node}" sh -c "echo $(docker inspect --format '{{.NetworkSettings.IPAddress }}' "${REGISTRY_CONTAINER_NAME}") registry >> /etc/hosts"
+done
+```
+
+## Using The Registry
+
+The registry can be used like this.
+
+1. First we'll pull an images `docker pull docker pull gcr.io/google-samples/hello-app:1.0`
```
Comment (aojea) [inline site/content/docs/user/local-registry.md]: double `docker pull` :upside_down_face:

File: site/content/docs/user/local-registry.md (hunk 4 above)
Comment (aojea) [reply site/content/docs/user/local-registry.md]: the command is wrong, it should something like `docker tag gcr.io/google-samples/hello-app:1.0 localhost:5000/hello-app:1.0`

File: site/content/docs/user/local-registry.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+2. Then we'll tag the image to use the local registry `docker tag gcr.io/google-samples/hello-app:1.0 localhost:5000`
+3. Then we'll push it to the registry `docker push localhost:5000`
```
Comment (aojea) [inline site/content/docs/user/local-registry.md]: ... and `docker push localhost:5000/hello-app:1.0`

File: site/content/docs/user/local-registry.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+4. And now we can run the image `kubectl create deployment hello-server --image=registry:5000/hello-app:1.0`
+
+If you build your own images and tag them with `localhost:5000/image:foo` and then use
+them in kubernetes as `registry:500/image:foo`.
```
Comment (aojea) [inline site/content/docs/user/local-registry.md]: maybe mention don't using `latest` as a tag?

**Comment by aojea** [comment]
I've tested locally with the comments I've added and it LGTM

File: site/content/docs/user/local-registry.md (hunk 7 above)
Comment (BenTheElder) [reply site/content/docs/user/local-registry.md]: 🤷‍♂

**Pull request merged**